### PR TITLE
Journalist can create articles

### PR DESCRIPTION
**Pull request by johanperjulius1**
## This PR contains: 
- Unit and integration tests
- Devise token auth
- Active Record enum
- Updated seed file

## [PT](https://www.pivotaltracker.com/story/show/175113656)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,8 @@
 class ApplicationController < ActionController::API
+  include DeviseTokenAuth::Concerns::SetUserByToken
+
+  def error_message(johan)
+    error_message = johan.full_messages.to_sentence
+    render json: { message: error_message }, status: 422
```
Comment (jennymalmefeldt) [inline app/controllers/application_controller.rb]: I like you Johan but why are your name in the controller? Is this a trick? 😉

**Review by ctl-lab — APPROVED** [review]
Just same as Jenny commented :)

File: db/migrate/20201014171538_add_journalist_to_articles.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddJournalistToArticles < ActiveRecord::Migration[6.0]
+  def change
+    add_column :articles, :journalist_id, :integer
```
Comment (oliverochman) [inline db/migrate/20201014171538_add_journalist_to_articles.rb]: you need to add this to the articles model

```
  belongs_to :journalist, class_name: 'User'

```

Comment (oliverochman) [inline db/migrate/20201014171538_add_journalist_to_articles.rb]: please refactor this to make sure it works as intended

`add_reference :articles, :journalist, foreign_key: { to_table: :users }`

File: app/controllers/api/v1/admin/articles_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def role_journalist?
+    unless current_user.role == "journalist"
+      restrict_access
+    end
+  end
+
+  def restrict_access
+    render json: { message: "Sorry, you don't have the necessary permission" }
```
Comment (oliverochman) [inline app/controllers/api/v1/admin/articles_controller.rb]: you need to test this sad path

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/admin/articles_controller.rb (hunk 3 above)
Comment (johanperjulius1) [reply app/controllers/api/v1/admin/articles_controller.rb]: Done!

File: db/migrate/20201014171538_add_journalist_to_articles.rb (hunk 2 above)
Comment (johanperjulius1) [reply db/migrate/20201014171538_add_journalist_to_articles.rb]: We updated it!

Comment (johanperjulius1) [reply db/migrate/20201014171538_add_journalist_to_articles.rb]: Done!

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (johanperjulius1) [reply app/controllers/application_controller.rb]: fixed!

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,2 +1,8 @@
 class ApplicationController < ActionController::API
+  include DeviseTokenAuth::Concerns::SetUserByToken
+
+  def error_message(johan)
+    error_message = johan.full_messages.to_sentence
```
Comment (emtalen) [inline app/controllers/application_controller.rb]: johan?

File: package.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "api_good_morning_news",
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "https://github.com/grconnor/api_good_morning_news.git",
+  "author": "<connorroelofsen15@gmail.com>",
+  "license": "MIT"
+}
```
Comment (emtalen) [inline package.json]: Why do you have a package.json?

**Review by emtalen — COMMENTED** [review]
You aslo need to add DS_Store to your .gitignore and clear the cache, because they have (accidentally) been pushed up

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (johanperjulius1) [reply app/controllers/application_controller.rb]: I thought I changed this before, jenny spotted it as well but somehow we missed to change it, will change it now!

File: package.json (hunk 5 above)
Comment (johanperjulius1) [reply package.json]: will remove!

**Review by emtalen — APPROVED** [review]

**Pull request merged**